docs/question/accessibility.md

```diff
@@ -0,0 +1,80 @@
+# Authoring accessible questions
+
+Ensuring that PrairieLearn questions are accessible to all students, including those with disabilities, is crucial for an inclusive learning environment. This guide provides recommendations for authors to create accessible question content.
+
+## Semantic HTML
+
+Using semantic HTML elements appropriately is the foundation of web accessibility. Semantic HTML means using HTML tags for their intended purpose, which helps assistive technologies (like screen readers) understand the structure and meaning of the content.
+
+- **Headings**: Use `<h1>`, `<h2>`, etc., to structure your question content logically. Avoid skipping heading levels.
+- **Lists**: Use `<ol>` for ordered lists and `<ul>` for unordered lists, and `<li>` for list items.
+- **Paragraphs**: Wrap text content in `<p>` tags.
+- **Emphasis**: Use `<em>` for emphasis and `<strong>` for strong importance, rather than relying on visual styling alone.
+
+## Text alternatives for non-text content
+
+All non-text content must have a text alternative that serves an equivalent purpose.
+
+### Images and figures (`pl-figure`)
+
+When using the [`pl-figure`](../elements.md#pl-figure-element) element to embed images, always provide descriptive alternative text (alt text) via the `alt` attribute. Good alt text is concise and conveys the essential information or function of the image.
+
+```html
+<pl-figure
+  file-name="diagram.png"
+  alt="A free-body diagram showing forces acting on a block on an inclined plane."
+></pl-figure>
+```
+
+If the figure contains complex information that's essential for understanding the content, you should provide that information elsewhere in the question text.
+
+### Input elements
+
+Many PrairieLearn elements that involve student input have attributes to enhance accessibility. For example, most input elements support a `label` attribute that is shown visually next to the input:
+
+```html
+<pl-integer-input answers-name="num_apples" label="Number of apples:"></pl-integer-input>
+```
+
+Sometimes, the visual label may not be sufficient for screen readers. In such cases, you can use the `aria-label` attribute to provide a more descriptive label that is not visible on the page but can be read by assistive technologies. For instance, in the following example, the visual label is "x =", but the screen reader will read "Enter the value of x in the equation 2x + 3 = 7":
```

Comment:
I'm a bit worried that readers will think that they actually need to use `aria-label` for math questions as shown here. Can we come up with an example that is either realistic or obviously fake?

Reply:
The actual feedback we got from UMich was that we should actually be using `aria-label` more, as tabbing to an input should be enough to identify what it is. However, I think their feedback was actually specifically about `<pl-multiple-choice>` and `<pl-checkbox>`, which currently just have a default label like "multiple choice options". This matters specifically for longer or multipart questions where students using a screen reader may get confused as to which input they're working with.

I'll adjust this example to use `<pl-multiple-choice>` instead.

Reply:
Changed in 9614a21.